pyproject.toml

```diff
@@ -0,0 +1,7 @@
+[tool.ruff]
+line-length=79
+[tool.ruff.lint.pycodestyle]
+max-line-length = 79
+[flake8]
+max-line-length = 79
+extend-ignore = W503
```

Comment:
flake8 is redundant with ruff. You should remove it completely from the picture.

Also, now that you use ruff for formatting, you **MUST** check that the formatting is consistent.